client/chain/chain.go

```diff
@@ -894,9 +946,19 @@ func (c *chainClient) GetSubAccountNonce(ctx context.Context, subaccountId eth.H
 	return c.exchangeQueryClient.SubaccountTradeNonce(ctx, req)
 }
 
+// Deprecated: Use CreateSpotOrder instead
 func (c *chainClient) SpotOrder(defaultSubaccountID eth.Hash, network common.Network, d *SpotOrderData) *exchangetypes.SpotOrder {
+	assistant, err := NewMarketsAssistant(network.Name)
```

Comment:
Although this is going to be deprecated, this can cause performance issue because we read file + run for-loop all the time, we should cache the read market assistant or initialize once

Reply:
Added logic in the markets_assistant.go module to make the legacy markets assistant (initialized using the .ini files) a singleton with lazy initialization (to not create it if the user is not using it)